.github/workflows/r_package.yml

```diff
@@ -48,12 +44,6 @@ jobs:
           ################
           # CMake builds #
           ################
-          - os: ubuntu-latest
-            task: r-package
-            compiler: gcc
-            r_version: 3.6
-            build_type: cmake
-            container: 'ubuntu:18.04'
```

Comment:
Trying to fix https://github.com/microsoft/LightGBM/pull/6733#issuecomment-2507125349, it seems that the workaround from https://github.com/actions/checkout/issues/1590#issuecomment-2207052044 is no longer working.

See https://github.com/actions/checkout/issues/1590#issuecomment-2507159983.

Between this and all the work that was required with package-pinning in #6718, I think we should just fully stop testing R 3.6 on Linux in CI here. As we did for macOS in #5859.

Note that we'll still have test coverage on Windows:

https://github.com/microsoft/LightGBM/blob/83c0ff3de1925b0e2d4831a9ccb6ffc196aa795b/.github/workflows/r_package.yml#L81-L87

https://github.com/microsoft/LightGBM/blob/83c0ff3de1925b0e2d4831a9ccb6ffc196aa795b/.github/workflows/r_package.yml#L95-L102

Dropping this testing gets us the following:

* reduces CI configuration complexity (see all the code removed in this PR)
* allows upgrading to `actions/checkout@v4` (no more skipping PRs like #6712)
* reduced ongoing maintenance burden, via avoiding things like these:
  - #6718
  - #6298
  - #5903